corehq/apps/accounting/tests/forms/test_trigger_customer_invoice_form.py

```diff
@@ -0,0 +1,234 @@
+
+    data = {
+        'month': month,
+        'year': year,
+        'customer_account': customer_account,
+        'num_users': num_users,
+        'overwrite_user_counts': overwrite_user_counts,
+    }
+
+    return TriggerCustomerInvoiceForm(data, show_testing_options=show_testing_options)
+
+
+class TestTriggerCustomerInvoiceFormValidity(SimpleTestCase):
+    TESTING_FIELDS = {'num_users', 'overwrite_user_counts'}
+
+    def test_has_all_details_is_valid_form(self):
+        form = create_form()
+        self.assertTrue(form.is_valid())
+
+    def test_missing_month_is_invalid(self):
+        form = create_form(month=None)
+        self.assertFalse(form.is_valid())
+
+    def test_missing_year_is_invalid(self):
+        form = create_form(year=None)
+        self.assertFalse(form.is_valid())
+
+    def test_missing_billing_account_is_invalid(self):
+        form = create_form(customer_account=None)
+        self.assertFalse(form.is_valid())
+
+    def test_current_invoice_periods_are_not_valid(self):
+        current_date = datetime.date(2020, 1, 10)
+        form = create_form(month=1, year=2020)
+        with mock.patch('corehq.apps.accounting.forms.datetime') as mock_datetime:
+            mock_datetime.date.today.return_value = current_date
+            self.assertFalse(form.is_valid())
+            self.assertIn('Statement period must be in the past', str(form.errors))
+
+    def test_if_testing_options_are_enabled_then_it_displays_test_fields(self):
+        form = create_form(show_testing_options=True)
+        self.assertTrue(self.TESTING_FIELDS.issubset(set(form.fields)))
+
+    def test_if_testing_options_are_disabled_then_it_hides_test_fields(self):
+        form = create_form(show_testing_options=False)
+        self.assertTrue(self.TESTING_FIELDS.isdisjoint(set(form.fields)))
+
+
+class TestTriggerCustomerInvoice(TestCase):
+    def setUp(self):
+        self.billing_account_obj = self.create_billing_account()
+        self.month = 10
+        self.year = 2020
+        self.start_date = datetime.date(self.year, self.month, 1)
+        self.end_date = datetime.date(self.year, self.month, 31)
+
+        self.domains = []
+
+    def tearDown(self):
+        for domain_obj in self.domains:
+            domain_obj.delete()
+
+    def create_form(self, **kwargs):
+        if 'customer_account' not in kwargs:
+            kwargs['customer_account'] = self.billing_account_obj.name
+        kwargs['month'] = self.month
+        kwargs['year'] = self.year
+        return create_form(**kwargs)
+
+    def create_billing_account(self, name=None, invoicing_plan=None):
+        self.account = BillingAccount.create_account_for_domain('test_domain', created_by='TEST')
+
+        dirty = False
+        if name:
+            self.account.name = name
+            dirty = True
+
+        if invoicing_plan:
+            self.account.invoicing_plan = invoicing_plan
+            dirty = True
+
+        if dirty:
+            self.account.save()
```

Comment:
Because code is read more than written, my focus is always on readability over brevity. If several attributes were added that all needed to be saved, I could then see creating an array of these properties and setting them in the way you listed, but I wouldn't do it through kwargs. The initial function is more readable because you know what properties are expected just by the definition, rather than having to wade through what is in the body. Because we're in a test class, I also try to avoid loops and abstractions.